unidler.py

```diff
@@ -6,36 +7,70 @@
 import ssl
 import sys
 
-# import kubernetes
-# import requests
+import kubernetes
+from kubernetes import client, config
+from kubernetes.client.models import (
+    V1beta1HTTPIngressPath,
+    V1beta1HTTPIngressRuleValue,
+    V1beta1IngressBackend,
+    V1beta1IngressRule,
+)
+
+
+IDLED = 'mojanalytics.xyz/idled'
+IDLED_AT = 'mojanalytics.xyz/idled-at'
+INGRESS_CLASS = 'kubernetes.io/ingress.class'
+UNIDLER = 'unidler'
 
 
 logging.basicConfig(level=os.environ.get('LOG_LEVEL', 'DEBUG'))
 log = logging.getLogger('unidler')
 
 
-class RequestHandler(BaseHTTPRequestHandler):
+def run(host='0.0.0.0', port=8080):
+    try:
+        config.load_incluster_config()
+    except:
+        config.load_kube_config()
 
-    def do_HEAD(self):
-        self.respond(200)
+    unidler = UnidlerServer((host, port), RequestHandler)
+    log.info(f'Unidler listening on {host}:{port}')
+    unidler.serve_forever()
+
+
+class UnidlerServer(ThreadingMixIn, HTTPServer):
```

Comment:
So, stupid question: this is an HTTP server which uses threads to handle requests right?

Reply:
Yep